src/rust/ide/lib/view/src/graph_editor/component/node/port/output.rs

```diff
@@ -324,9 +325,8 @@ enum ShapeView {
 
 impl ShapeView {
     /// Constructor.
-    fn new(number_of_ports:NonZeroU32, logger:&Logger, scene:&Scene) -> Self {
-        let number_of_ports = number_of_ports.get();
-        if number_of_ports == 1 {
+    fn new(number_of_ports:u32, logger:&Logger, scene:&Scene) -> Self {
+        if number_of_ports <= 1 {
             ShapeView::Single { view: component::ShapeView::new(&logger,&scene) }
```

Comment:
This needs a description in doc comment to this function now (that you output Single) if port count was 0